### PR TITLE
Update Backstage Customization Docs to use ThemeProvider

### LINKED ISSUE #8648: Object literal may only specify known properties, and 'theme' does not exist in type 'Partial<AppTheme> & Omit<AppTheme, "theme">'
When adding a custom theme to App.tsx as documented here: https://backstage.io/docs/getting-started/app-custom-theme i should be able to make use of my custom theme -

```
const app = createApp({
  apis,
  plugins: [badgesPlugin],
  themes: [{
    id: 'customTheme',
    title: 'Custom',
    variant: 'dark',
    theme: customTheme,
 }],
  components: {
    SignInPage:
      props => {
        return (
          <SignInPage {...props}
           Guest to be used only for development purposes
           providers={[...providers]}
            align="center" />
        );
      },
  },
```
I am getting the following error once i upgraded to release-2021-12-23

```
 error TS2322: Type '{ id: string; title: string; variant: "dark"; theme: BackstageTheme; }' is not assignable to type 'Partial<AppTheme> & Omit<AppTheme, "theme">'.
  Object literal may only specify known properties, and 'theme' does not exist in type 'Partial<AppTheme> & Omit<AppTheme, "theme">'.
```

### PR DESCRIPTION
**Pull request by cmpadden**
closes #8648 

Documentation was outdated and instructed that a custom theme should be specified using the `theme` property instead of through a theme `Provider`.

I also included the imports for defining `customTheme`, and updated color codes to something with appropriate contrast over _123456_.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/getting-started/app-custom-theme.md [hunk 1]
```diff
@@ -76,60 +82,67 @@ want to use the default themes, they are exported as `lightTheme` and
...
 ```ts
-const themeOptions = createThemeOptions({
+import {
+  createTheme,
+  genPageTheme,
+  lightTheme,
+  shapes,
+} from '@backstage/theme';
+
+const customTheme = createTheme({
```
Comment (cmpadden) [inline docs/getting-started/app-custom-theme.md:92]: This was previously `createThemeOptions`, is that fine?

Comment (Rugvip) [reply docs/getting-started/app-custom-theme.md:92]: Yeah that seems like a more useful example. Could call it `myTheme` to make it line up with the example above

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

The overrides section just below is also outdated, prolly best to handle that one too while at it

File: docs/getting-started/app-custom-theme.md [hunk 2]
```diff
@@ -200,22 +200,22 @@ export const createCustomThemeOverrides = (
   };
 };
 
+const overrides = createCustomThemeOverrides(lightTheme);
+
+const overriddenTheme = { ...lightTheme, overrides };
```
Comment (cmpadden) [inline docs/getting-started/app-custom-theme.md:205]: @Rugvip , Please let me know if this was done incorrectly; I took reference from `packages/theme/src/baseTheme.ts` on how overrides were applied.

Comment (Rugvip) [reply docs/getting-started/app-custom-theme.md:205]: I think it should be like it was tbh

Comment (cmpadden) [reply docs/getting-started/app-custom-theme.md]: Updated! For some reason I thought I couldn't do it that way, but it was a silly mistake.

I've confirmed that the overriding the theme per the latest docs works on my instance.

**Comment by cmpadden** [comment]
Ah! just a heads-up, it looks like we're doing a similar thing in this PR  #8684!

**Comment by Rugvip** [comment]
@cmpadden oh lawd, well I'm gonna merge this and then bring in the stuff that's good from the other one by handling the merge conflict 😅

**Review by Rugvip — APPROVED** [review]

**Pull request merged**